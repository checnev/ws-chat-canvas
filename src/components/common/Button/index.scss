.button {
  padding: 8px 12px;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
  transition: all .25s ease;

  &_color_primary {
    border-color: rgb(30, 167, 253);
    color: rgb(30, 167, 253);
    background: rgb(255, 255, 255);

    &:hover {
      color: rgb(255, 255, 255);
      background: rgb(30, 167, 253);
    }
  }
}
