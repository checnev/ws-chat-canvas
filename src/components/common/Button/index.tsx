import React from 'react';
import cn from 'utils/cn';
import clsx from 'clsx';
import './index.scss';

export type Color = 'primary' | 'second';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>{
  color?: Color;
}

class Button extends React.PureComponent<ButtonProps> {
  static defaultProps: ButtonProps = {
    color: 'primary',
  };

  render() {
    const button = cn('button');
    const { className, color, ...props } = this.props;
    
    return (
      <button
        className={
          clsx(className, button({ color }))
        }
        {...props}
      />
    );
  }
}

export default Button;
