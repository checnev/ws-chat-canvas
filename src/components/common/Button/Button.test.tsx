import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Button from 'components/common/Button';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render Button', () => {
  act(() => {
    render(<Button>Button Text</Button>, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
