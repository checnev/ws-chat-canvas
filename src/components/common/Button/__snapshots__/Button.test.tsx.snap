// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button 1`] = `"<button class=\\"button button_color_primary\\">Button Text</button>"`;
